event/src/main/java/com/bazaarvoice/emodb/event/db/astyanax/AstyanaxEventReaderDAO.java

```diff
@@ -313,6 +342,25 @@ public void readNewer(String channel, EventSink sink) {
                 }
             }
         }
+
+        if (firstSlab && oldestSlab == null) {
+            // Channel was completely empty.  Cache a TimeUUID for the current time.  This will cause future calls
+            // to read at most 1 minute of tombstones until the cache expires 10 seconds later.
+            cacheOldestSlabForChannel(channel, TimeUUIDs.newUUID());
+        }
+    }
+
+    private void cacheOldestSlabForChannel(String channel, UUID slabId) {
+        // Functionally the same as ConcurrentMap.computeIfAbsent(...)
+        try {
+            // Subtract 1 minute from the slab ID to allow for a reasonable window of out-of-order writes while
+            // constraining the number of tombstones read to 1 minute's worth of rows.
+            _oldestSlab.get(channel, () ->
```

Comment:
@billkalter should it be put() here instead if get() :?

Reply:
Agreed, this is very confusing and I tried to put a comment to clarify.  With the newer Java interfaces like `ConcurrentMap` you can use `computeIfAbsent()` or `putIfAbsent()`.  The Guava cache interface doesn't have a similar method, but if you do a `get()` this way it does the same thing: caches the new value only if there is no current un-expired version in the cache.  From their docs:

> This method provides a simple substitute for the conventional "if cached, return; otherwise create, cache and return" pattern.